Core/src/Core/Filters/Algorithms/CalculateArrayHistogram.cpp

```diff
@@ -0,0 +1,315 @@
+#include "CalculateArrayHistogram.hpp"
+
+#include "Core/Filters/CalculateArrayHistogramFilter.hpp"
+
+#include "complex/Common/Range.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataGroup.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
+
+#include <chrono>
+
+using namespace complex;
+
+namespace
+{
+template <typename T>
+class GenerateHistogramFromData
+{
+public:
+  GenerateHistogramFromData(CalculateArrayHistogram& filter, const int32 numBins, const DataArray<T>& inputArray, Float64Array& histogram, std::atomic<usize>& overflow, float64 minRange = 0.0,
+                            float64 maxRange = 0.0)
+  : m_Filter(filter)
```

Comment:
I would be very tempted to combine the "Use Custom Range" with the actual range values and pass in a [std::tuple](https://en.cppreference.com/w/cpp/utility/tuple) and pass this is as:
```
std::tuple<bool, float, float>& customRange
```

This will allow the code calling this constructor to drop down to 
```        dataAlg.execute(GenerateHistogramFromData(*this, numBins, m_DataStructure.getDataRefAs<DataArray<int8>>(selectedArrayPath), histogram, overflow, {m_InputValues->UserDefinedRange, m_InputValues->MinRange, m_InputValues->MaxRange}));
```
and log of duplicated code can be removed